ui/ui_statusbar.c

```diff
@@ -13,111 +13,122 @@
 #include "display/scope.h"
 
 
-uint32_t ui_statusbar_info(char *buf)
+uint32_t ui_statusbar_info(char *buf, size_t buffLen)
 {
     uint32_t len=0;
     uint32_t temp=0;
     uint32_t cnt=0;
 
-	len+=ui_term_color_text_background_buf(&buf[len], 0x000000, BP_COLOR_GREY);
+	len+=ui_term_color_text_background_buf(&buf[len], buffLen, 0x000000, BP_COLOR_GREY);
+    buffLen = sizeof(tx_sb_buf) - len;	
     
     if(system_config.psu)
     {
-        temp=sprintf(&buf[len],"Vout: %u.%uV",
+        temp=snprintf(&buf[len], buffLen, "Vout: %u.%uV",
            (system_config.psu_voltage)/10000, ((system_config.psu_voltage)%10000)/100
         );
         len+=temp;
         cnt+=temp;
+        buffLen = sizeof(tx_sb_buf) - len;	

```

Comment:
I think this line needs to be removed.  

Because the first argument to `snprintf()` here takes the original buffer pointer, and offsets into it by `len`, it makes sense to "mirror" that for the buffer length argument by passing `buffLen - len`.